1/contrib/jenkins/kube-slave-common.sh

```diff
@@ -13,7 +13,15 @@
 
 export DEFAULT_SLAVE_DIRECTORY=/tmp
 export SLAVE_LABEL="jenkins-slave"
-export JENKINS_JNLP_SERVICE_PORT=${JENKINS_JNLP_SERVICE_PORT:-50000}
+JNLP_SERVICE_NAME=${JNLP_SERVICE_NAME:-JENKINS_JNLP}
+T=${JNLP_SERVICE_NAME}_SERVICE_HOST
+# the '!' handles env variable indirection so we can resolve the nested variable
+# see: http://stackoverflow.com/a/14204692
+JNLP_HOST=${!T}
+T=${JNLP_SERVICE_NAME}_SERVICE_PORT
```

Comment:
Just a nit, why not use a different variable name instead of relying on the sequence of things for T to have the right value?